compiler/testSuite/semtype/xml-complex-ro-t.bal

```diff
@@ -0,0 +1,41 @@
+type X xml;
+type RX readonly & X;
+
+type N xml<never>;
+type T xml:Text;
+type E readonly & xml:Element;
+type P readonly & xml:ProcessingInstruction;
+type C readonly & xml:Comment;
+type XE xml<E>;
+type XP xml<P>;
+type XC xml<C>;
+
+type S T|E|P|C;
+
+// @type NonEmptyS < S
+// @type NonEmptyS <> T
+// @type NonEmptyS <> N
+// @type E < NonEmptyS
+// @type P < NonEmptyS
+// @type C < NonEmptyS
+type NonEmptyS S & !N;
+
+// @type NonEmptyS < UX
+type UX XE|XP|XC|T;
+
+// @type XNonEmptyS = RX
+// @type XNonEmptyS < X
```

Comment:
Don't understand this one: `xml<T>` means sequence of length 0 or more where every member belongs to T. So I would say: XNonEmptyS = X. What am I missing?

Reply:
I improved the names reflect the type.
Here since T is `readonly`, xml<T> can only contain readonly parts hence it's a subtype of `X`

Reply:
OK. Got it.